module/input/SensorFilter/src/MotionModel.hpp

```diff
@@ -181,32 +170,30 @@ namespace module::input {
             return newState;
         }
 
-        Eigen::Matrix<Scalar, 3, 1> predict(const Eigen::Matrix<Scalar, size, 1>& state,
-                                            const MeasurementType::ACCELEROMETER&) {
-            // Extract our rotation quaternion
-            const Eigen::Matrix<Scalar, 3, 3> Rtw = StateVec(state).Rwt.inverse().toRotationMatrix();
-
-            // Make a world gravity vector and rotate it into torso space
-            // Where is world gravity with respect to robot orientation?
-            // Multiplying a matrix with (0, 0, G) is equivalent to taking the
-            // third column of the matrix and multiplying it by G
-            return Rtw.template rightCols<1>() * G;
+        Eigen::Matrix<Scalar, 3, 1> predict(const StateVec& state, const MeasurementType::ACCELEROMETER&) {
+
+            // Rotate world gravity vector into torso space using quaternion conjugation
+            // p' = q * p * q.conjugate()
+            // G' = Rtw * G * Rtw.conjugate()
+            //    = Rwt.conjugate() * G * Rwt.conjugate().conjugate()
+            //    = Rwt.conjugate() * G * Rwt
+            // G is a quaternion with real part zero and vector part (0, 0, G)
+            // The vector part of G' is the result of rotating G by Rtw
```

Comment:
This explanation says a lot of things, but it doesn't leave me feeling that I understand what the line is doing.
I think this _might_ be better split across a few lines, with play-by-play commentary as it happens? idk

Reply:
It's literally a 1-line equation? I'm not sure how to break this up anymore than it already is. I am literally just laying out the math

Reply:
maybe construct this: `Eigen::Quaternion<Scalar>(Scalar(0), Scalar(0), Scalar(0), Scalar(G))`, commenting what it represents, then do the rest.